### PR TITLE
Fix the issue of Add/Update webhook destination with null method, add…

### PR DESCRIPTION
**Pull request by Abhishek-121**
…ing log message for subscription with site not present,  Error while addign syslog server in fresh set up as in latest sdk exception msg got change.

## Description
Please include a summary of the changes and the related issue. Also, include relevant motivation and context.

-- Fix the following issues in the events and notification module -
   
     - Add/Update webhook destination with null method should return specific error statement
     - Error while addign syslog server in the fresh set up as in latest dnacentersdk the exception message got changed.
     - Update the log message for create subscription with site not present.

-- Also add the check in the module that it start supporting from catalyst center version 2.3.5.3 onwards.

## Type of Change
- [ yes ] Bug fix
- [ no ] New feature
- [ no ] Breaking change
- [ no ] Documentation update

## Checklist
- [ yes ] My code follows the style guidelines of this project
- [ yes ] I have performed a self-review of my own code
- [ yes ] I have commented my code, particularly in hard-to-understand areas
- [ yes ] I have made corresponding changes to the documentation
- [ yes ] My changes generate no new warnings
- [ no ] I have added tests that prove my fix is effective or that my feature works
- [ no ] New and existing unit tests pass locally with my changes
- [ no ] Any dependent changes have been merged and published in downstream modules
- [ yes ] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [ yes ] Tasks are idempotent (can be run multiple times without changing state)
- [ no ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [ yes ] Playbooks are modular and reusable
- [ no ] Handlers are used for actions that need to run on change

## Documentation
- [ yes ] All options and parameters are documented clearly.
- [ yes ] Examples are provided and tested.
- [ yes ] Notes and limitations are clearly stated.

### CODE REVIEW

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 1]
```diff
@@ -1262,12 +1262,15 @@ def get_syslog_destination_in_ccc(self, name):
             return response
 
         except Exception as e:
-            if "Expecting value: line 1 column 1" in str(e):
-                self.log(
-                    "Getting expection as Syslog destination with given name '{0}' not present in Cisco Catalyst"
-                    " Center.".format(name), "WARNING"
-                )
-                return None
+            expected_exception_msg = ["Expecting value: line 1 column 1", "not iterable", "has no attribute"]
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:1265]: can we write one by one for more readability?

```
expected_exception_msgs = [
    "Expecting value: line 1 column 1",
    "not iterable",
    "has no attribute"
]

```

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 2]
```diff
@@ -1262,12 +1262,15 @@ def get_syslog_destination_in_ccc(self, name):
...
+            for msg in expected_exception_msg:
+                if msg in str(e):
+                    self.log(
+                        "Getting expection as Syslog destination with given name '{0}' not present in Cisco Catalyst"
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:1269]: "An exception occurred while checking for the Syslog destination with the name '{0}'. It was not found in Cisco Catalyst Center.".format(name),

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 3]
```diff
@@ -1500,12 +1503,14 @@ def get_snmp_destination_in_ccc(self, name):
                             return destination
                     time.sleep(1)
                 except Exception as e:
-                    if "Expecting value: line 1 column 1" in str(e):
-                        self.log(
-                            "Getting expection as SNMP destination with given name '{0}' not present in Cisco Catalyst"
-                            " Center.".format(name), "WARNING"
-                        )
-                        return None
+                    expected_exception_msg = ["Expecting value: line 1 column 1", "not iterable", "has no attribute"]
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:1506]: same as above..

Can you please rephrase the log message?

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 4]
```diff
@@ -1850,14 +1855,17 @@ def get_webhook_destination_in_ccc(self, name):
...
+
                     time.sleep(1)
                 except Exception as e:
-                    if "Expecting value: line 1 column 1" in str(e):
-                        self.log(
-                            "Getting expection as Syslog destination with given name '{0}' not present in Cisco Catalyst"
-                            " Center.".format(name), "WARNING"
-                        )
-                        return None
+                    expected_exception_msg = ["Expecting value: line 1 column 1", "not iterable", "has no attribute"]
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:1861]: same as above

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 5]
```diff
@@ -2119,12 +2127,15 @@ def get_email_destination_in_ccc(self):
             return response[0]
 
         except Exception as e:
-            if "Expecting value: line 1 column 1" in str(e):
-                self.log(
-                    "Getting expection as Email destination is not configured in Cisco Catalyst"
-                    " Center.", "WARNING"
-                )
-                return None
+            expected_exception_msg = ["Expecting value: line 1 column 1", "not iterable", "has no attribute"]
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:2130]: same as above

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 6]
```diff
@@ -2759,12 +2775,15 @@ def get_syslog_notification_details(self, name):
             return response
 
         except Exception as e:
-            if "Expecting value: line 1 column 1" in str(e):
-                self.log(
-                    "Getting expection as Syslog Event Notification with given name '{0}' not present in Cisco Catalyst"
-                    " Center.".format(name), "WARNING"
-                )
-                return None
+            expected_exception_msg = ["Expecting value: line 1 column 1", "not iterable", "has no attribute"]
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:2778]: same as above

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 7]
```diff
@@ -2806,12 +2825,15 @@ def get_syslog_subscription_detail(self, destination):
             return response[0]
 
         except Exception as e:
-            if "Expecting value: line 1 column 1" in str(e):
-                self.log(
-                    "Getting expection as Syslog destination with given name '{0}' not present in Cisco Catalyst"
-                    " Center.".format(destination), "WARNING"
-                )
-                return None
+            expected_exception_msg = ["Expecting value: line 1 column 1", "not iterable", "has no attribute"]
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:2828]: same as above

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 8]
```diff
@@ -3137,8 +3159,11 @@ def collect_syslog_notification_playbook_params(self, syslog_notification_detail
         if sites:
             site_ids = self.get_site_ids(sites)
             if not site_ids:
-                self.msg = "Unable to find the Site ID's for the given site(s) - '{0}' in the playbook's input.".format(sites)
-                self.log(self.msg, "INFO")
+                site_msg = (
+                    "Unable to find the Site IDs for the given site(s) - '{0}' in the playbook's input for the "
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:3163]: "No Site IDs were found for the specified site(s) - '{0}' in the playbook input during the "
    "Syslog event notification operation."

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 9]
```diff
@@ -3527,12 +3552,15 @@ def get_webhook_notification_details(self, name):
             return response
 
         except Exception as e:
-            if "Expecting value: line 1 column 1" in str(e):
-                self.log(
-                    "Getting expection as Webhook Event Notification with given name '{0}' not present in Cisco Catalyst"
-                    " Center.".format(name), "WARNING"
-                )
-                return None
+            expected_exception_msg = ["Expecting value: line 1 column 1", "not iterable", "has no attribute"]
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:3555]: same as above

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 10]
```diff
@@ -3569,12 +3597,15 @@ def get_webhook_subscription_detail(self, destination):
             return response[0]
 
         except Exception as e:
-            if "Expecting value: line 1 column 1" in str(e):
-                self.log(
-                    "Getting expection as Webhook destination with given name '{0}' not present in Cisco Catalyst"
-                    " Center.".format(destination), "WARNING"
-                )
-                return None
+            expected_exception_msg = ["Expecting value: line 1 column 1", "not iterable", "has no attribute"]
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:3600]: same as above

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 11]
```diff
@@ -3684,8 +3715,11 @@ def collect_webhook_notification_playbook_params(self, webhook_notification_deta
         if sites:
             site_ids = self.get_site_ids(sites)
             if not site_ids:
-                self.msg = "Unable to find the Site IDs for the given site(s) - '{0}' in the playbook's input.".format(sites)
-                self.log(self.msg, "INFO")
+                site_msg = (
+                    "Unable to find the Site IDs for the given site(s) - '{0}' in the playbook's input for the "
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:3719]: "No Site IDs were found for the specified site(s) - '{0}' in the playbook input during the "
    "Webhook event notification operation."

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 12]
```diff
@@ -3965,12 +3999,15 @@ def get_email_notification_details(self, name):
             return response
 
         except Exception as e:
-            if "Expecting value: line 1 column 1" in str(e):
-                self.log(
-                    "Getting expection as Email Event Notification with given name '{0}' not present in Cisco Catalyst"
-                    " Center.".format(name), "WARNING"
-                )
-                return None
+            expected_exception_msg = ["Expecting value: line 1 column 1", "not iterable", "has no attribute"]
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:4002]: same as above

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 13]
```diff
@@ -4008,12 +4045,15 @@ def get_email_subscription_detail(self, instance):
             return response[0]
 
         except Exception as e:
-            if "Expecting value: line 1 column 1" in str(e):
-                self.log(
-                    "Getting expection as Email instance with given name '{0}' not present in Cisco Catalyst"
-                    " Center.".format(instance), "WARNING"
-                )
-                return None
+            expected_exception_msg = ["Expecting value: line 1 column 1", "not iterable", "has no attribute"]
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:4048]: same as above

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 14]
```diff
@@ -4188,8 +4228,11 @@ def collect_email_notification_playbook_params(self, email_notification_details)
         if sites:
             site_ids = self.get_site_ids(sites)
             if not site_ids:
-                self.msg = "Unable to find the Site IDs for the given site(s) - '{0}' in the playbook's input.".format(sites)
-                self.log(self.msg, "INFO")
+                site_msg = (
+                    "Unable to find the Site IDs for the given site(s) - '{0}' in the playbook's input for the "
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:4232]: "No Site IDs were found for the specified site(s) - '{0}' in the playbook input during the "
    "Email event notification operation."

File: plugins/modules/events_and_notifications_workflow_manager.py [hunk 15]
```diff
@@ -5474,6 +5517,14 @@ def main():
         ccc_events.check_return_status()
 
     ccc_events.validate_input().check_return_status()
+    if ccc_events.get_ccc_version_as_integer() <= ccc_events.get_ccc_version_as_int_from_str("2.3.5.3"):
```
Comment (madhansansel) [inline plugins/modules/events_and_notifications_workflow_manager.py:5520]: if ccc_events.compare_dnac_versions(ccc_sda_devices.get_ccc_version(), "2.3.5.3") < 0:
        ccc_events.msg = (
            "The specified version '{0}' does not support the events and notifications workflow. Supported versions start from '2.3.5.3' onwards."
            .format(ccc_events.get_ccc_version())
        )
        ccc_events.set_operation_result("failed", False, ccc_events.msg, "ERROR").check_return_status()

Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 14 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 13 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 12 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 11 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 10 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 9 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 8 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 7 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 6 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 5 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 4 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 3 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 2 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

File: plugins/modules/events_and_notifications_workflow_manager.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/events_and_notifications_workflow_manager.py]: updated.

**Review by madhansansel — APPROVED** [review]

**Pull request merged**